webodm.sh

```diff
@@ -510,17 +510,26 @@ run_tests(){
     # If in a container, we run the actual test commands
     # otherwise we launch this command from the container
     if [[ -f /.dockerenv ]]; then
-        echo -e "\033[1mRunning frontend tests\033[0m"
-        run "npm run test"
-
-        echo "\033[1mRunning backend tests\033[0m"
-        run "python manage.py test"
-
-        echo ""
-        echo -e "\033[1mDone!\033[0m Everything looks in order."
+        test_type=${1:-"all"}
+        shift || true
+        
+        if [[ $test_type = "frontend" || $test_type = "all" ]]; then
+            echo -e "\033[1mRunning frontend tests\033[0m"
+            run "npm run test $@"
+        fi
+
+        if [[ $test_type = "backend" || $test_type = "all" ]]; then
+            echo -e "\033[1mRunning backend tests\033[0m"
+            run "python manage.py test $@"
+        fi
+
+        if [[ $test_type = "all" ]]; then
+            echo ""
+            echo -e "\033[1mDone!\033[0m Everything looks in order."
+        fi
     else
         echo "Running tests in webapp container"
-        run "$docker_compose exec webapp /bin/bash -c \"/webodm/webodm.sh test\""
+        run "$docker_compose exec webapp /bin/bash -c \"/webodm/webodm.sh test $@\""
```

Comment:
There seems to be some problem with the expansion of the $@ variable here:

```bash
[pt:~/Documents/WebODM] selective_tests(+2/-1) 1 ± ./webodm.sh test all app.tests.TestApi
Running tests in webapp container
docker-compose exec webapp /bin/bash -c "/webodm/webodm.sh test all
./webodm.sh: eval: line 348: unexpected EOF while looking for matching `"'
```

Reply:
[pt:~/Documents/WebODM_clean] selective_tests(+1/-1) ± bash --version
GNU bash, version 3.2.57(1)-release (arm64-apple-darwin24)
Copyright (C) 2007 Free Software Foundation, Inc.

Reply:
Replicated it and (hopefully) solved.